app/models/concerns/requestable.rb

```diff
@@ -7,7 +7,13 @@ module Requestable
     belongs_to :assessment
 
     enum :state,
-         { requested: "requested", received: "received", expired: "expired" },
+         {
+           requested: "requested",
+           received: "received",
+           expired: "expired",
+           accepted: "accepted",
+           rejected: "rejected",
```

Comment:
I was thinking about how to model this, and I wondered if we needed to capture this as two separate states:

- `state`: `requested`, `received`, `expired`
- `review_state`: `unknown`, `accepted`, `rejected`

My reasoning is that if the state is `accepted` that means it must also be `received`, and we've got logic in a few places that checks if the request has been `received?` but this'll be `false` if the state is `accepted` (or `rejected`).

https://github.com/DFE-Digital/apply-for-qualified-teacher-status/blob/141a8080aec38d347acba5dac5cc281afbb84435/app/lib/application_form_status_updater.rb#L126-L128

We could update the various places where we check for `received?` to also include `accepted?` and `rejected?` but I'd be worried about forgetting to do that in a certain place, or just generally it feels right that something could be `received?` and `accepted?` at the same time.

With further information requests, we model this as a nullable boolean `passed`, and I wonder if we can simply move this out of further information request and in to the `Requestable` concern so all requestables have the same mechanism.

Reply:
Ah OK good call. I was going on the assumption that `accepted` and `rejected` were implicitly `received` but it's not as explicit as having 2 different state mechanisms. I'll look at integrating with `Requestable`.